test/integration.test.ts

```diff
@@ -284,6 +284,70 @@ const testCases: {
       expect(stderr).toContain(log)
     },
   },
+  {
+    name: 'external-entry',
+    args: ['--external', 'external-entry'],
+    async expected(dir) {
+      const distFiles = [
+        join(dir, './dist/index.cjs'),
+        join(dir, './dist/index.mjs'),
+        join(dir, './dist/server.cjs'),
+        join(dir, './dist/server.mjs'),
+      ]
+
+      expect(await fs.readFile(distFiles[0], 'utf-8')).toContain(
+        `Large piece of code`,
+      )
+      expect(await fs.readFile(distFiles[1], 'utf-8')).toContain(
+        `Large piece of code`,
+      )
+      expect(await fs.readFile(distFiles[2], 'utf-8')).not.toContain(
+        `Large piece of code`,
+      )
+      expect(await fs.readFile(distFiles[3], 'utf-8')).not.toContain(
+        `Large piece of code`,
+      )
+
+      expect(execSync('node ' + distFiles[2]).toString()).toBe(
+        `Large piece of code\n`,
+      )
+      expect(execSync('node ' + distFiles[3]).toString()).toBe(
+        `Large piece of code\n`,
+      )
+    },
+  },
+  {
+    name: 'external-index',
+    args: ['--external', './index'],
```

Comment:
`external-entry` itself will be added to external by default, it shouldn't require any `args`

Reply:
Depending on export conditions, that could be tricky though right?

Reply:
Maybe interesting to show a warning that an external thaty starts with `./` is not supported?

Reply:
It should be supported, I've reached to the rollup-plugin-swc3 maintainer to help investigate the module resolving